swap/tests/e2e.rs

```diff
@@ -200,16 +242,61 @@ async fn happy_path_recursive_executor() {
             v_a,
         }
     };
-    let alice_swarm =
-        alice::new_swarm(alice_multiaddr.clone(), alice_transport, alice_behaviour).unwrap();
-    let config = xmr_btc::config::Config::regtest();
-    let alice_swap = alice::swap::swap(
+
+    let alice_swarm = alice::new_swarm(alice_multiaddr, alice_transport, alice_behaviour).unwrap();
+
+    (
         alice_state,
         alice_swarm,
-        alice_btc_wallet.clone(),
-        alice_xmr_wallet.clone(),
-        config,
+        alice_btc_wallet,
+        alice_xmr_wallet,
+        alice_peer_id,
+    )
+}
+
+#[allow(clippy::too_many_arguments)]
+async fn init_bob(
+    alice_multiaddr: Multiaddr,
+    alice_peer_id: PeerId,
+    bitcoind: &Bitcoind<'_>,
+    monero: &Monero,
+    btc_to_swap: bitcoin::Amount,
+    btc_starting_balance: bitcoin::Amount,
+    xmr_to_swap: u64,
+    xmr_stating_balance: u64,
+) -> (
+    BobState,
+    bob::Swarm,
+    Arc<swap::bitcoin::Wallet>,
+    Arc<swap::monero::Wallet>,
+    Database,
+) {
+    let bob_btc_wallet = Arc::new(
+        swap::bitcoin::Wallet::new("bob", bitcoind.node_url.clone())
+            .await
+            .unwrap(),
     );
+    bitcoind
+        .mint(
+            bob_btc_wallet.0.new_address().await.unwrap(),
+            btc_starting_balance,
+        )
+        .await
+        .unwrap();
+
+    monero
+        .init(vec![("bob", xmr_stating_balance)])
+        .await
+        .unwrap();
+
+    let bob_xmr_wallet = Arc::new(swap::monero::Wallet(monero.wallet("bob").unwrap().client()));
+
+    let amounts = SwapAmounts {
+        btc: btc_to_swap,
+        xmr: xmr_btc::monero::Amount::from_piconero(xmr_to_swap),
+    };
+
+    let rng = &mut OsRng;
```

Comment:
You seem to only use this var once, I would not name it, just pass `OsRng` directly.